src/bin/background-migration/index.ts

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env node -r sucrase/register
+
+import childProcess from 'child_process'
+import { Cleaner } from 'cleaners'
+import { access, mkdir, readFile, writeFile } from 'fs/promises'
+import fetch, { Response } from 'node-fetch'
+import { join, sep as pathSeperator } from 'path'
+import { promisify } from 'util'
+
+import { config } from './config'
+import {
+  asCheckpoint,
+  asScanFile,
+  asScanIndex,
+  Checkpoint,
+  ScanFile,
+  ScanIndex
+} from './types'
+
+const exec = promisify(childProcess.exec)
+
+// Config
+const { dataDir, sshHost, syncServer, migrationEndpoint } = config
+
+// ---------------------------------------------------------------------
+// App
+// ---------------------------------------------------------------------
+
+async function main(): Promise<void> {
+  await setup()
+
+  // Get index
+  const scanIndex = await getScanIndex()
+
+  // Get checkpoint
+  let checkpoint: Checkpoint | null = await getCheckpoint(scanIndex.partitions)
+
+  while (checkpoint !== null) {
+    const scanFile = await getScanFile(checkpoint.partition)
+    const { repoIds } = scanFile
+
+    for (let index = checkpoint.index; index < repoIds.length; ++index) {
+      const repoId = repoIds[index]
+      const res = await migrateRepo(repoId)
+      const status = res.status
+
+      console.log(
+        `Migrated ${repoId} (partition ${checkpoint.partition} at index ${index})`
+      )
+
+      if (status < 200 || status >= 300) {
+        const body = await res.json()
+        console.error(`${repoId} ${status} response:`, body)
```

Comment:
Add the failed repo to an append only failed-repos.log file. The log format should be easy to parse for later